redbot/core/core_commands.py

```diff
@@ -3351,10 +3351,16 @@ async def _set_ownernotifications_listdestinations(self, ctx: commands.Context):
     # -- End Set Ownernotifications Commands -- ###
 
     @_set.command(name="showsettings")
-    async def _set_showsettings(self, ctx: commands.Context):
+    async def _set_showsettings(
+        self, ctx: commands.Context, specified_guild: discord.Guild = None
+    ):
         """
-        Show the current settings for [botname].
+        Show the current settings for [botname]. Accepts optional guild parameter in case must recover prefix.
         """
+        if specified_guild is not None:
+            # override ctx.guild if a guild is provided as an argument
+            ctx.guild = specified_guild
+
```

Comment:
These changes should not be a part of this PR. This command is only for viewing the **current** settings.

Reply:
Right, that makes sense. Added changes to use server variable rather than modifying ctx.guild as mentioned below.